app/src/test/java/com/otaman/notes/viewmodel/AllNotesViewModelTest.kt

```diff
@@ -0,0 +1,64 @@
+package com.otaman.notes.viewmodel
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.otaman.notes.model.FakeNoteRepository
+import com.otaman.notes.model.Note
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.*
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+internal class AllNotesViewModelTest {
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @get:Rule
+    val mainDispatcherRule = MainDispatcherRule()
+
+    private lateinit var fakeNoteRepository: FakeNoteRepository
+    private lateinit var allNotesViewModel: AllNotesViewModel
+    private val note = Note(title = "Title1", content = "Content1")
+    private val note1 = Note(title = "Title2", content = "Content2")
+
+    @Before
+    fun setup() {
+        fakeNoteRepository = FakeNoteRepository()
+        allNotesViewModel = AllNotesViewModel(fakeNoteRepository)
+    }
+
+    @Test
+    fun deleteNote_deletesNote() = runTest {
+        allNotesViewModel.deleteNote(note)
```

Comment:
To test this function properly you should add some notes to the view model and them delete one. Then yous should  test that only the note with the requested id was actually removed